src/metabase/api/database.clj

```diff
@@ -713,6 +724,87 @@
+                                              (if schema
+                                                (str schema "." table-name)
+                                                table-name))})))
+      (first tables))))
+
+(defn- resolve-table [db-id table-identifier]
+  (let [[schema table-name] (let [parts (str/split table-identifier #"\.")]
+                              (case (count parts)
+                                1 [nil (first parts)]
+                                2 parts
+                                nil))]
+    (api/check table-name 400 "Invalid table identifier")
+    (api/check-404
+     (or (check-unique (t2/select :model/Table
+                                  :db_id db-id
+                                  :name table-name
+                                  (cond-> {}
+                                    schema (assoc :where {:schema schema}))))
+         (check-unique (t2/select :model/Table
+                                  :db_id db-id
+                                  :%lower.name (u/lower-case-en table-name)
+                                  (cond-> {}
+                                    schema (assoc :where {:%lower.schema (u/lower-case-en schema)}))))))))
+
+(defn- remap-202->200 [response]
+  (cond
+    ;; Well, this is pretty intrusive. Unfortunately, there's no way to set it correctly in the first place, yet.
+    (and (instance? StreamingResponse response) (contains? #{202 nil} (:status (.-options response))))
+    (metabase.server.streaming-response/->StreamingResponse
+     (.-f response)
+     (assoc (.-options response) :status 200)
+     (.-donechan response))
+
+    (and (map? response) (contains? #{202 nil} :status-code response))
+    (assoc response :status-code 200)
+
+    ;; ¯\\_(ツ)_/¯
+    :else
+    response))
+
+(api.macros/defendpoint :get "/:db-id/table/:table-identifier"
+  "Get the data for the given table"
+  [{:keys [db-id table-identifier]} :- [:map
+                                        [:db-id ms/PositiveInt]
+                                        [:table-identifier ms/NonBlankString]]]
+  (api/read-check (t2/select-one :model/Database :id db-id))
+  (qp.store/with-metadata-provider db-id
+    (let [mp    (qp.store/metadata-provider)
+          table (resolve-table db-id table-identifier)
+          query (-> (lib/query mp (lib.metadata/table mp (:id table)))
+                    (update-in [:middleware :js-int-to-string?] (fnil identity true))
+                    qp/userland-query-with-default-constraints
+                    (update :info merge {:executed-by api/*current-user-id*
+                                         :context     :table-grid
+                                         :card-id     nil}))]
+      ;; TODO Discuss whether we want these analytics
+      #_(events/publish-event! :event/table-read {:object  (t2/select-one :model/Table :id table-id)
+                                                  :user-id api/*current-user-id*})
+      (remap-202->200
```

Comment:
but why do we care about changing 202 to 200?

Reply:
Just being a good web standards person. "Accepted" seems like a crazy code for a GET.

Yes, the effort is totally ridiculous, and we don't want to merge it like this. I was thinking to refactor the `streaming-response` method so we could inject the default success code as an option. Currently you can only override by implementing a multimethod for the "type" (e.g. `:api`)

AFAIK we still return 5xx codes if there's some kind of usual error. I guess there's still a possibility of say an IO error once we start streaming though.

Reply:
Going to just back this out, there are much better eyebrow raisers in our APIs, and refactoring to make this clean seems like a waste of time in this phase of the project.